backend/compact-connect/stacks/persistent_stack/user_email_notifications.py

```diff
@@ -0,0 +1,74 @@
+from aws_cdk.aws_iam import ServicePrincipal
+from aws_cdk.aws_ses import EmailIdentity, Identity, ConfigurationSet, EmailSendingEvent, EventDestination
+from aws_cdk.aws_sns import Topic, Subscription, SubscriptionProtocol
+from aws_cdk.aws_route53 import TxtRecord, IHostedZone
+from aws_cdk.aws_kms import IKey
+
+from constructs import Construct
+
+
+class UserEmailNotifications(Construct):
+    """
+    This Construct leverages SES to set up an email notification system to send cognito user events from our custom
+    domain with necessary SPF, DKIM, and DMARC verification records.
+
+    The topic is set up to forward all bounce and complaint events to the provided email address.
+    """
+    def __init__(
+            self, scope: Construct, construct_id: str, *,
+            hosted_zone: IHostedZone,
+            environment_context: dict,
+            master_key: IKey,
+            **kwargs
+    ):
+        super().__init__(scope, construct_id, **kwargs)
+
+        domain_name = hosted_zone.zone_name
+        operation_email = environment_context['notifications']['ses_feedback_forwarding_email']
+
+
+        self.email_feedback_topic = Topic(self, "FeedbackTopic",
+                                          display_name="Email Feedback Forwarding Topic",
+                                          master_key=master_key,
+                                          enforce_ssl=True,
+                                          )
+        Subscription(self, "FeedbackSubscription",
+                     topic=self.email_feedback_topic,
+                     protocol=SubscriptionProtocol.EMAIL,
+                     endpoint=operation_email)
+
+        self.config_set = ConfigurationSet(self, "ConfigSet")
+
+        self.config_set.add_event_destination(id="EmailFeedbackEventDestination",
+                                         destination=EventDestination.sns_topic(self.email_feedback_topic),
+                                         configuration_set_event_destination_name="EmailFeedbackDestination",
+                                         enabled=True,
+                                         events=[EmailSendingEvent.BOUNCE,
+                                                 EmailSendingEvent.COMPLAINT]
+                                         )
+
+        ses_principal = ServicePrincipal("ses.amazonaws.com")
+        self.email_feedback_topic.grant_publish(ses_principal)
+        # grant SES the ability to encrypt bounce and complaint notifications using the KMS key
+        master_key.grant_encrypt_decrypt(ses_principal)
+
+
+        # Create SES Email Identity with DKIM enabled
+        self.email_identity = EmailIdentity(self, "EmailIdentity",
+            # by using the hosted zone resource, cdk will automatically
+            # create all the necessary DNS records for DKIM and SPF authentication
+            identity=Identity.public_hosted_zone(hosted_zone),
+            mail_from_domain=f"no-reply.{domain_name}",
```

Comment:
Can you talk me through the benefit of using this custom from domain?

Reply:
discussed f2f, my understanding is that if you are going to use a sub domain for your from address (ie no-reply) this is needed for CDK to generate the needed TXT and MX records specifically to verify the use of that subdomain.